In [1]:
def bayes_theorem(p_a, p_b_given_a, p_b_given_not_a):
    # 計算檢測結果呈陽性的概率
    #貝斯定理
    p_b = (p_b_given_a * p_a) + (p_b_given_not_a * (1 - p_a))
    
    # 計算後驗機率
    p_a_given_b = (p_b_given_a * p_a) / p_b
    
    return p_a_given_b

# 這裡需要提供具體的數據
# 例如，先驗機率 P(A)、在有癌症的情況下檢測為陽性的條件概率 P(B|A)->p_b_given_a
# 在沒有癌症的情況下檢測為陽性的條件概率 P(B|¬A)-> p_b_given_not_a 
p_a = 0.02  # 先驗機率(代表國家多少人羅癌的比例)
p_b_given_a = 0.97  # 在有癌症的情況下檢測為陽性的條件概率
p_b_given_not_a = 0.09  # 在沒有癌症的情況下檢測為陽性的條件概率

result = bayes_theorem(p_a, p_b_given_a, p_b_given_not_a)
print(f"患病的後驗機率為: {result}")


患病的後驗機率為: 0.18029739776951673


0.02 * (0.97/0.09) 

In [26]:
0.97 * 0.02  / ((0.97 * 0.02) + (0.09 * 0.98))

0.18029739776951673

全國有2%的人有罹癌 * 97%真實是陽性的 / (兩種機率的總和(0.97 * 0.02 + 0.09 *0.98)) 
其中的0.97 * 0.02 是有癌症檢測陽性的機率，0.09 * 0.98是無癌症檢測陽性的機率
算出來剛好是0.18029739 = 18%